refinery_core/src/runner.rs

```diff
@@ -399,3 +414,80 @@ impl Runner {
         .await
     }
 }
+
+/*impl<C> IntoIterator for Runner
+where
+    C: Migrate,
+{
+    type Item = Result<Migration, Error>;
+    type IntoIter = RunIterator;
+
+    fn into_iter(self) -> Self::IntoIter {
+        Runner::into_iter(self)
+    }
+}*/
+
+pub struct RunIterator<'a, C> {
+    connection: &'a mut C,
+    target: Target,
+    migration_table_name: String,
+    items: VecDeque<Migration>,
+    failed: bool,
+}
+impl<'a, C> RunIterator<'a, C>
+where
+    C: Migrate,
+{
+    pub(crate) fn new(runner: Runner, connection: &'a mut C) -> RunIterator<'a, C> {
+        RunIterator {
+            items: VecDeque::from(
+                Migrate::get_unapplied_migrations(
+                    connection,
+                    &runner.migrations,
+                    runner.abort_divergent,
+                    runner.abort_missing,
+                    &runner.migration_table_name,
+                )
+                .unwrap(),
+            ),
+            connection,
+            target: runner.target,
+            migration_table_name: runner.migration_table_name.clone(),
+            failed: false,
+        }
+    }
+}
+impl<C> Iterator for RunIterator<'_, C>
+where
+    C: Migrate,
+{
+    type Item = Result<Migration, Error>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.failed {
+            true => None,
+            false => self.items.pop_front().and_then(|migration| {
+                sync_migrate(
+                    self.connection,
+                    vec![migration],
+                    self.target,
+                    &self.migration_table_name,
+                )
+                .map(|r| r.applied_migrations.first().cloned())
+                .map_err(|e| {
+                    error!("migration failed: {e:?}");
+                    self.failed = true;
+                    e
+                })
+                .transpose()
+            }),
+        }
+    }
+}
+
+fn flatten<T>(o: Option<Option<T>>) -> Option<T> {
```

Comment:
is this function used? If so we can use [`flatten`](https://doc.rust-lang.org/std/option/enum.Option.html#method.flatten) right instead?

Reply:
ha, no, this was a workaround for another approach that did use flatten (which is still unstable somehow), but it's dead code now